darkhttpd.c

```diff
@@ -2291,6 +2291,33 @@ static void process_get(struct connection *conn) {
     }
 }
 
+/* Returns 1 if passwords are equal, runtime is proportional to the length of
+ * user_input to avoid leaking the secret's length and contents through timing
+ * information.
+ */
+int password_equal(const char *user_input, const char *secret) {
+    size_t i = 0;
+    size_t j = 0;
+    char out = 0;
+
+    while (1) {
+        /* Out stays zero if the strings are the same. */
+        out |= user_input[i] ^ secret[j];
+
+        /* Stop at end of user_input. */
+        if (user_input[i] == 0) break;
```

Comment:
Hi folks, could providing a huge password (GBs; parallel requests) lead to DoS? I saw [this line](https://github.com/emikulic/darkhttpd/blob/a4a4fcd1e9229c11d44cda3222ed137ee45c223b/README.md?plain=1#L38) and found [this check](https://github.com/emikulic/darkhttpd/blob/a4a4fcd1e9229c11d44cda3222ed137ee45c223b/darkhttpd.c#L2404). So, I suppose the, rather theoretical, DoS is handled.